ast/ast.go

```diff
@@ -816,11 +816,12 @@ func (n *StringNode) String() string {
 		// It works mostly, but inconsistencies occur if line break characters are mixed.
 		header := token.LiteralBlockHeader(n.Value)
 		space := strings.Repeat(" ", n.Token.Position.Column-1)
+		indent := strings.Repeat(" ", n.Token.Position.IndentNum)
```

Comment:
The token's IndentNum represents the number of spaces from the beginning of the line. In other words, this is not a value that represents "how many spaces are used for indentation".

Therefore, the behavior is different between the case where the Go value is directly converted to a string using the encoder, and the case where the string is created using the ast obtained by parsing.

However, when encoding the value of Go directly, the context `literal` is missing, so the AST that is always created becomes a `StringNode`, whereas when parsing, it becomes a `LiteralNode`. For this reason, treating the value of `IndentNum` in this way works well for `StringNode`.

This behavior may cause confusion, so it may be necessary to refactor it in the future.